ogre/src/OgreWideAngleCamera.cc

```diff
@@ -0,0 +1,662 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifdef __APPLE__
+  #define GL_SILENCE_DEPRECATION
+  #include <OpenGL/gl.h>
+  #include <OpenGL/glext.h>
+#else
+#ifndef _WIN32
+  #include <GL/gl.h>
+  #include <GL/glext.h>
+
+  // OGRE/RenderSystems/GL/GL/glext.h does not seem to have
+  // this defined
+  #ifndef GL_TEXTURE_CUBE_MAP_SEAMLESS
+    #define GL_TEXTURE_CUBE_MAP_SEAMLESS 0x884F
+  #endif
+#endif
+#endif
+
+#include "ignition/rendering/CameraLens.hh"
+
+#include "ignition/rendering/ogre/OgreConversions.hh"
+#include "ignition/rendering/ogre/OgreRenderEngine.hh"
+#include "ignition/rendering/ogre/OgreRTShaderSystem.hh"
+#include "ignition/rendering/ogre/OgreWideAngleCamera.hh"
+
+/// \brief Private data for the WideAngleCamera class
+class ignition::rendering::OgreWideAngleCameraPrivate
+{
+  /// \brief Environment texture size
+  public: int envTextureSize = 512u;
+
+  /// \brief Compositor used to render rectangle with attached cube map
+  public: Ogre::CompositorInstance *cubeMapCompInstance = nullptr;
+
+  /// \brief A Set of 6 cameras,
+  ///   each pointing in different direction with FOV of 90deg
+  public: Ogre::Camera *envCameras[6];
+
+  /// \brief Render targets for envCameras
+  public: Ogre::RenderTarget *envRenderTargets[6];
+
+  /// \brief Viewports for the render targets
+  public: Ogre::Viewport *envViewports[6];
+
+  /// \brief Pixel format for cube map texture
+  public: Ogre::PixelFormat envCubeMapTextureFormat;
+
+  /// \brief A single cube map texture
+  public: Ogre::Texture *envCubeMapTexture = nullptr;
+
+  /// \brief Output texture
+  public: Ogre::Texture *ogreRenderTexture = nullptr;
+
+  /// \brief Pointer to material, used for second rendering pass
+  public: Ogre::MaterialPtr compMat;
```

Comment:
```suggestion
  public: Ogre::MaterialPtr compMat{nullptr};
```

Reply:
this is an `Ogre::SharedPtr`, which unlike `std::shared_ptr, can not be set to `nullptr`. So I kept it the way it is.